### PR TITLE
Closes issue #25

### PR DESCRIPTION
**Pull request by builat**
* Closes issue #25 
* Resolves button styled override
* Fixes some perf issues
* Fixes button types interface

### CODE REVIEW

File: src/components/button/README.md [hunk 1]
```diff
@@ -19,11 +19,8 @@ interface Props {
 ## Button types
 
 ```typescript
-export enum ButtonType {
-  default = "default",
-  noFill = "noFill",
-  borderless = "borderless",
-}
+  type ButtonType = "default" | "noFill" | "borderless",
```
Comment (VLegakis) [inline src/components/button/README.md]: Nope. Why is this required to change?

File: src/components/button/styled.tsx [hunk 2]
```diff
@@ -176,7 +131,7 @@ export const StyledButton = styled(({ label, icon, ...otherProps }) => (
     opacity: ${props.disabled ? 0.6 : 1.0};
     cursor: ${props.disabled ? "not-allowed" : "pointer"};
     background-color: ${buttonProps("buttonColorNormal", props)};
-    border-color: ${getColor(["green", "greenHaze"])}
+    border-color: ${getGreenHaze}
```
Comment (Azein) [inline src/components/button/styled.tsx]: Easier to change with just getColor, let's not write a special named function to get a certain color.

File: src/components/button/button.tsx [hunk 3]
```diff
@@ -25,17 +21,18 @@ export const Button: FC<ButtonProps> = ({
   onClick = () => {},
   icon = null,
   isLoading = false,
-  type = ButtonType.default,
+  type = DEFAULT,
```
Comment (VLegakis) [inline src/components/button/button.tsx]: Why take this from a string constants file instead of the original enum?

File: src/components/button/button.tsx [hunk 4]
```diff
@@ -25,17 +21,18 @@ export const Button: FC<ButtonProps> = ({
...
   disabled = false,
+  ...rest
```
Comment (Azein) [inline src/components/button/button.tsx]: let's use ...props

File: src/components/button/README.md [hunk 5]
```diff
@@ -44,16 +41,10 @@ const myButton: Button = (
...
 ```typescript
 const myButton: Button = (
-  <Button
-    disabled={true}
-    type={ButtonTypes.default}
-    icon={"plus"}
-    onClick={this.myHandler}
-    label="Press me"
-  />
+  <Button disabled={true} type="default" icon={"plus"} onClick={this.myHandler} label="Press me" />
```
Comment (Azein) [inline src/components/button/README.md]: I suggest to use icon the same way as in <TextInput>, thus passing the component, not a string.

File: src/components/button/button.tsx [hunk 6]
```diff
@@ -25,17 +21,18 @@ export const Button: FC<ButtonProps> = ({
   onClick = () => {},
```
Comment (Azein) [inline src/components/button/button.tsx]: Let's mark returning type as void, () => {} seems to be a bit misleading

File: src/components/button/styled.tsx [hunk 7]
```diff
@@ -2,127 +2,82 @@ import React from "react"
...
+const getGreenMalachite = getColor(["green", "malachite"])
 
 const buttonPropsMap = new Map<string, (props: ButtonProps) => any>([
-  [
-    "divWidth",
-    props => {
-      return props.label ? "134px" : "46px"
-    },
-  ],
+  ["divWidth", ({ label }) => (label ? "134px" : "46px")],
```
Comment (Azein) [inline src/components/button/styled.tsx]: px should be removed in favour of getSizeBy / other means of getting size from theme-defined constant.

File: src/components/button/README.md (hunk 1 above)
Comment (Azein) [reply src/components/button/README.md]: As I understood, the inconvenience is that you have to import enum from the component to set the type other than default. Correct me if I'm wrong. If so, I'm 100% for this change, you should not be obliged to import enum types from UI component module to be able to set some customization.

Comment (builat) [reply src/components/button/README.md]: Why should I do
```typescript
import {Button, ButtonTypes } from '@netdata/netdat-ui'
function Component() {
  return(<>
    <Button label="any" type={ButtonType.noFill} .../>
  </>)
}
```
Instead of just "noFill". A lot of redundant extra complexity

File: src/components/button/button.tsx (hunk 4 above)
Comment (builat) [reply src/components/button/button.tsx]: done

File: src/components/button/button.tsx (hunk 6 above)
Comment (jacekkolasa) [reply src/components/button/button.tsx]: @Azein i think it is void, it's not type definition, it's the default value here
